### PR TITLE
EVG-3218 Create list hosts endpoint

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: rest/data/creathost.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+// DBCreateHostConnector supports `create.host` commands from the agent.
+type DBCreateHostConnector struct{}
+
+// ListHostsForTask lists running hosts scoped to the task or the task's build.
+func (*DBCreateHostConnector) ListHostsForTask(taskID string) ([]model.CreateHost, error) {
+	t, err := task.FindOneId(taskID)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding task")
```
Comment (john-m-liu) [inline rest/data/creathost.go]: return rest.APIError so that you can set the status code

File: rest/route/createhost_routes.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		return ResponseData{}, rest.APIError{
+			StatusCode: http.StatusInternalServerError,
+			Message:    "error listing hosts for task",
+		}
+	}
+	results := make([]model.Model, len(hosts))
+	for i := range hosts {
+		results[i] = model.Model(&hosts[i])
+	}
+	return results, nil
```
Comment (john-m-liu) [inline rest/route/createhost_routes.go]: Compile issue here

**Comment by tychoish** [comment]
if you rebase and are feeling adventurous you could implement this using the new stuff.

File: rest/data/creathost.go (hunk 1 above)
Comment (bsamek) [reply rest/data/creathost.go]: fixed

File: rest/route/createhost_routes.go (hunk 2 above)
Comment (bsamek) [reply rest/route/createhost_routes.go]: fixed

**Pull request merged**